chips/esp32-c3/src/timg.rs

```diff
@@ -0,0 +1,244 @@
+
+        (0x070 => int_ena: ReadWrite<u32, INT::Register>),
+        (0x074 => int_raw: ReadWrite<u32, INT::Register>),
+        (0x078 => int_st: ReadWrite<u32, INT::Register>),
+        (0x07C => int_clr: ReadWrite<u32, INT::Register>),
+        (0x080 => @END),
+    }
+}
+
+register_bitfields![u32,
+    CONFIG [
+        USE_XTAL OFFSET(9) NUMBITS(1) [],
+        ALARM_EN OFFSET(10) NUMBITS(1) [],
+        LEVEL_INT_EN OFFSET(11) NUMBITS(1) [],
+        DIVIDER_RST OFFSET(12) NUMBITS(1) [],
+        DIVIDER OFFSET(13) NUMBITS(16) [],
+        AUTORELOAD OFFSET(29) NUMBITS(1) [],
+        INCREASE OFFSET(30) NUMBITS(1) [],
+        EN OFFSET(31) NUMBITS(1) [],
+    ],
+    WDTCONFIG0 [
+        FLASHBOOT_MOD_EN OFFSET(14) NUMBITS(1) [],
+        SYS_RESET_LENGTH OFFSET(15) NUMBITS(3) [],
+        CPU_RESET_LENGTH OFFSET(18) NUMBITS(3) [],
+        LEVEL_INT_EN OFFSET(21) NUMBITS(1) [],
+        EDGE_INT_EN OFFSET(22) NUMBITS(1) [],
+        STG3 OFFSET(23) NUMBITS(2) [],
+        STG2 OFFSET(25) NUMBITS(2) [],
+        STG1 OFFSET(27) NUMBITS(2) [],
+        STG0 OFFSET(29) NUMBITS(2) [],
+        EN OFFSET(31) NUMBITS(1) [],
+    ],
+    WDTCONFIG1 [
+        CLK_PRESCALE OFFSET(16) NUMBITS(16) [],
+    ],
+    RTCCALICFG [
+        START_CYCLING OFFSET(12) NUMBITS(1) [],
+        CLK_SEL OFFSET(13) NUMBITS(2) [],
+        RDY OFFSET(15) NUMBITS(1) [],
+        MAX OFFSET(16) NUMBITS(15) [],
+        START OFFSET(31) NUMBITS(1) [],
+    ],
+    RTCCALICFG1 [
+        VALUE OFFSET(7) NUMBITS(25) [],
+    ],
+    INT [
+        T0 OFFSET(0) NUMBITS(1) [],
+        WDT OFFSET(1) NUMBITS(1) [],
+    ],
+];
+
+#[repr(u8)]
+#[derive(Copy, Clone)]
+pub enum ClockSource {
+    Pll = 0,
+    Xtal = 1,
+}
+
+pub struct TimG<'a> {
+    registers: StaticRef<TimgRegisters>,
+    clocksource: ClockSource,
+    alarm_client: OptionalCell<&'a dyn time::AlarmClient>,
+}
+
+impl TimG<'_> {
+    pub const fn new(base: StaticRef<TimgRegisters>, clocksource: ClockSource) -> Self {
+        TimG {
+            registers: base,
+            clocksource,
+            alarm_client: OptionalCell::empty(),
+        }
+    }
+
+    pub fn handle_interrupt(&self) {
+        self.registers.int_clr.modify(INT::T0::SET);
+        let _ = self.stop();
+        self.alarm_client.map(|client| {
+            client.alarm();
+        });
+    }
+
+    pub fn disable_wdt(&self) {
+        self.registers
+            .wdtconfig0
+            .modify(WDTCONFIG0::EN::CLEAR + WDTCONFIG0::FLASHBOOT_MOD_EN::CLEAR);
+
+        if self.registers.wdtconfig0.is_set(WDTCONFIG0::EN)
+            || self
+                .registers
+                .wdtconfig0
+                .is_set(WDTCONFIG0::FLASHBOOT_MOD_EN)
+        {
+            panic!("Can't disable TIMG WDT");
+        }
+    }
+}
+
+impl time::Time for TimG<'_> {
+    type Frequency = Freq20MHz;
+    type Ticks = Ticks64;
+
+    fn now(&self) -> Self::Ticks {
+        self.registers.t0update.set(0xABC);
```

Comment:
what is `0xABC` here? A comment or named constant would be helpful

Reply:
My understanding is that you have to write anything in that register, the value does not matter. I did not write this part, I think @alistair23 can answer this better. I'll add a comment.